crates/containerd-shim-wasm/src/sandbox/shim/otel.rs

```diff
@@ -0,0 +1,164 @@
+//! OpenTelemetry Configuration Module
+//!
+//! This module provides a configuration structure and associated methods to initialize
+//! OpenTelemetry tracing with the OTLP exporter. The configuration can be set up via
+//! the `OtelConfig` struct and its builder pattern.
+//!
+//! # Usage
+//!
+//! ```rust
+//! use containerd-shim-wasm::sandbox::shim::otel::{OtelConfig, OTEL_EXPORTER_OTLP_ENDPOINT};
+//!
+//! fn main() -> anyhow::Result<()> {
+//!     let otel_endpoint = std::env::var(OTEL_EXPORTER_OTLP_ENDPOINT).expect("OTEL_EXPORTER_OTLP_ENDPOINT not set");
+//!     let otel_config = OtelConfig::builder()
+//!         .otel_endpoint(otel_endpoint)
+//!         .name("my-service".to_string())
+//!         .build()?;
+//!
+//!     let _guard = otel_config.init()?;
+//!
+//!     // Your application code here
+//!
+//!     Ok(())
+//! }
+//! ```
+
+use std::collections::HashMap;
+
+use opentelemetry::global::{self, set_text_map_propagator};
+use opentelemetry::trace::TraceError;
+use opentelemetry_otlp::{
+    SpanExporterBuilder, WithExportConfig, OTEL_EXPORTER_OTLP_PROTOCOL_DEFAULT,
+};
+pub use opentelemetry_otlp::{OTEL_EXPORTER_OTLP_ENDPOINT, OTEL_EXPORTER_OTLP_PROTOCOL};
+use opentelemetry_sdk::propagation::TraceContextPropagator;
+use opentelemetry_sdk::{runtime, trace as sdktrace};
+use tracing::Span;
+use tracing_opentelemetry::OpenTelemetrySpanExt as _;
+use tracing_subscriber::layer::SubscriberExt as _;
+use tracing_subscriber::{EnvFilter, Registry};
+
+const OTEL_EXPORTER_OTLP_PROTOCOL_HTTP_PROTOBUF: &str = "http/protobuf";
```

Comment:
These are already defined here: https://docs.rs/opentelemetry-otlp/latest/src/opentelemetry_otlp/exporter/mod.rs.html#48

Reply:
Yeah but that's not a public facing const (cannot use it outside of the crate)

Reply:
Ah right! I guess I'm a bit rusty! 🦀